dsc/workflows/base/__init__.py

```diff
@@ -392,16 +314,29 @@ def item_submissions_iter(self) -> Iterator[ItemSubmission]:
         MUST NOT be overridden by workflow subclasses.
         """
         for item_metadata in self.item_metadata_iter():
+            self.submission_summary["total"] += 1
             item_identifier = self.get_item_identifier(item_metadata)
+            try:
+                item_submission = ItemSubmission.from_batch_id_and_item_identifier(
+                    batch_id=self.batch_id, item_identifier=item_identifier
+                )
+            except DoesNotExist:
```

Comment:
I originally had ambitions of using the new `ItemSubmission.get_batch_items()` method to start instantiating the item_submissions but it got complicated so holding off until you both can chime

Reply:
I'm good with this for now, if we are assuming this will potentially get reworked in the submit ticket [IN-1356](https://mitlibraries.atlassian.net/browse/IN-1356).  

I could be misunderstanding something, but it feels like we'd want to drive that submit loop from the Dynamo rows, full stop.  We reconciled, all looks good, so we have Dynamo DB rows basically outlining the items in the batch.  

It feels like you tucked in enough here to keep it functional -- and used that convenience method `.from_batch_id_and_item_identifier()`! -- but left the overall control flow the same.  

This PR is already an undertaking, agree that waiting makes sense.

Reply:
Once the DSpace metadata  functionality shifts into `ItemSubmission`, I think this will be easier to modify. That can be done after this!